From 1948a0b34b2478424bb00afc46032a92aecb7b6c Mon Sep 17 00:00:00 2001
From: Hanyuan Zhao <hanyuan-z@qq.com>
Date: Wed, 18 Oct 2023 20:57:20 +0800
Subject: [PATCH 7/7] fix: add more sub set relocate handlers

Signed-off-by: Hanyuan Zhao <hanyuan-z@qq.com>
---
 arch/riscv/kernel/module.c | 58 +++++++++++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/arch/riscv/kernel/module.c b/arch/riscv/kernel/module.c
index 7605e6628..e4d854a99 100644
--- a/arch/riscv/kernel/module.c
+++ b/arch/riscv/kernel/module.c
@@ -270,6 +270,28 @@ static int apply_r_riscv_add64_rela(struct module *me, u32 *location,
 	return 0;
 }
 
+static int apply_r_riscv_sub6_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u8 *)location -= ((u8)v & 0x3f);
+	*(u8 *)location &= 0x3f;
+	return 0;
+}
+
+static int apply_r_riscv_sub8_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u8 *)location -= (u8)v;
+	return 0;
+}
+
+static int apply_r_riscv_sub16_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u16 *)location -= (u16)v;
+	return 0;
+}
+
 static int apply_r_riscv_sub32_rela(struct module *me, u32 *location,
 				    Elf_Addr v)
 {
@@ -284,6 +306,35 @@ static int apply_r_riscv_sub64_rela(struct module *me, u32 *location,
 	return 0;
 }
 
+static int apply_r_riscv_set6_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u8 *)location &= 0xc0;
+	*(u8 *)location |= (u8)v & 0x3f;
+	return 0;
+}
+
+static int apply_r_riscv_set8_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u8 *)location = (u8)v;
+	return 0;
+}
+
+static int apply_r_riscv_set16_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u16 *)location = (u16)v;
+	return 0;
+}
+
+static int apply_r_riscv_set32_rela(struct module *me, u32 *location,
+				    Elf_Addr v)
+{
+	*(u32 *)location = (u32)v;
+	return 0;
+}
+
 static int apply_r_riscv_pcrel_32_rela(struct module *me, u32 *location,
 				    Elf_Addr v)
 {
@@ -313,8 +364,15 @@ static int (*reloc_handlers_rela[]) (struct module *me, u32 *location,
 	[R_RISCV_ALIGN]			= apply_r_riscv_align_rela,
 	[R_RISCV_ADD32]			= apply_r_riscv_add32_rela,
 	[R_RISCV_ADD64]			= apply_r_riscv_add64_rela,
+	[R_RISCV_SUB6]			= apply_r_riscv_sub6_rela,
+	[R_RISCV_SUB8]			= apply_r_riscv_sub8_rela,
+	[R_RISCV_SUB16]			= apply_r_riscv_sub16_rela,
 	[R_RISCV_SUB32]			= apply_r_riscv_sub32_rela,
 	[R_RISCV_SUB64]			= apply_r_riscv_sub64_rela,
+	[R_RISCV_SET6]			= apply_r_riscv_set6_rela,
+	[R_RISCV_SET8]			= apply_r_riscv_set8_rela,
+	[R_RISCV_SET16]			= apply_r_riscv_set16_rela,
+	[R_RISCV_SET32]			= apply_r_riscv_set32_rela,
 	[R_RISCV_32_PCREL]		= apply_r_riscv_pcrel_32_rela,	
 };
 
-- 
2.34.1

